pytensor/tensor/rewriting/linalg.py

```diff
@@ -37,87 +63,98 @@ def inv_as_solve(fgraph, node):
     """
     if isinstance(node.op, (Dot, Dot22)):
         l, r = node.inputs
-        if l.owner and isinstance(l.owner.op, MatrixInverse):
+        if (
+            l.owner
+            and isinstance(l.owner.op, Blockwise)
+            and isinstance(l.owner.op.core_op, MatrixInverse)
+        ):
             return [solve(l.owner.inputs[0], r)]
-        if r.owner and isinstance(r.owner.op, MatrixInverse):
+        if (
+            r.owner
+            and isinstance(r.owner.op, Blockwise)
+            and isinstance(r.owner.op.core_op, MatrixInverse)
+        ):
             x = r.owner.inputs[0]
             if getattr(x.tag, "symmetric", None) is True:
-                return [solve(x, l.T).T]
+                return [T(solve(x, T(l)))]
             else:
-                return [solve(x.T, l.T).T]
+                return [T(solve(T(x), T(l)))]
 
 
 @register_stabilize
 @register_canonicalize
-@node_rewriter([Solve])
+@node_rewriter([Blockwise])
 def tag_solve_triangular(fgraph, node):
     """
     If a general solve() is applied to the output of a cholesky op, then
     replace it with a triangular solve.
 
     """
-    if isinstance(node.op, Solve):
-        if node.op.assume_a == "gen":
+    if isinstance(node.op.core_op, Solve) and node.op.core_op.b_ndim == 1:
+        if node.op.core_op.assume_a == "gen":
             A, b = node.inputs  # result is solution Ax=b
-            if A.owner and isinstance(A.owner.op, Cholesky):
-                if A.owner.op.lower:
-                    return [Solve(assume_a="sym", lower=True)(A, b)]
-                else:
-                    return [Solve(assume_a="sym", lower=False)(A, b)]
             if (
                 A.owner
-                and isinstance(A.owner.op, DimShuffle)
-                and A.owner.op.new_order == (1, 0)
+                and isinstance(A.owner.op, Blockwise)
+                and isinstance(A.owner.op.core_op, Cholesky)
```

Comment:
This becomes messy if there are blockwise ops and non blockwise ops under consideration.

Reply:
Rewrites now never work for non-blockwise. We expect users to always work with blockwise (the same way we always expect users to work with Elemwise and not scalar Ops)